library/std/src/thread/mod.rs

```diff
@@ -499,13 +499,18 @@ impl Builder {
         let output_capture = crate::io::set_output_capture(None);
         crate::io::set_output_capture(output_capture.clone());
 
+        // Pass `f` in `MaybeUninit` because actually that closure might *run longer than the lifetime of `F`*.
+        // See <https://github.com/rust-lang/rust/issues/101983> for more details.
+        let f = mem::MaybeUninit::new(f);
```

Comment:
Hm, this has the side effect that if the closure is dropped before being run, we leak `f`. That is probably not what we want...

Reply:
No I don't think it does.